django/db/backends/oracle/operations.py

```diff
@@ -217,7 +218,14 @@ def deferrable_sql(self):
         return " DEFERRABLE INITIALLY DEFERRED"
 
     def fetch_returned_insert_id(self, cursor):
-        return int(cursor._insert_id_var.getvalue())
+        try:
+            return int(cursor._insert_id_var.getvalue())
+        except TypeError:
+            raise DatabaseError(
+                'The database did not return a new row id. Probably "ORA-1403:'
```

Comment:
```
'The database did not return a new row id. Probably "ORA-1403: '
'no data found" was raised internally but was hidden by the '
'Oracle OCI library (see https://code.djangoproject.com/ticket/28859).'
```
I don't like including URLs in error messages, but in this case it seems okay as users may not know how to convert the ticket number into a URL for the bug tracker.